src/header/common/link.rs

```diff
@@ -0,0 +1,622 @@
+    /// service
+    Service,
+    /// start
+    Start,
+    /// stylesheet
+    Stylesheet,
+    /// subsection
+    Subsection,
+    /// successorVersion
+    SuccessorVersion,
+    /// up
+    Up,
+    /// versionHistory
+    VersionHistory,
+    /// via
+    Via,
+    /// working-copy
+    WorkingCopy,
+    /// working-copy-of
+    WorkingCopyOf,
+    /// ext-rel-type
+    ExtRelType(Uri)
+}
+
+impl Link {
+    /// Create Link from URI-Reference
+    pub fn new(uri: &str) -> ::Result<Link> {
+        match Uri::new(uri) {
+            Err(_) => Err(::Error::Header),
+            Ok(u) => Ok(Link {
+                link: u,
+                rel: None,
+                anchor: None,
+                rev: None,
+                href_lang: Vec::new(),
+                media_desc: None,
+                title: None,
+                title_star: None,
+                media_type: None,
+                link_extension: None,
+            })
+        }
+    }
+}
+
+impl Header for Link {
+    fn header_name() -> &'static str {
+        static NAME: &'static str = "Link";
+        NAME
+    }
+
+    fn parse_header(raw: &Raw) -> ::Result<Link> {
+       from_one_raw_str(raw)
+    }
+
+    fn fmt_header(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // TODO: Implementation missing.
+        write!(f, "<{}>", self.link)
+    }
+}
+
+impl fmt::Display for Link {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.fmt_header(f)
+    }
+}
+
+impl FromStr for Link {
+    type Err = ::Error;
+
+    fn from_str(s: &str) -> ::Result<Link> {
+        let mut link_split = s.split(';');
+
+        // Parse the `Target IRI`
+        // https://tools.ietf.org/html/rfc5988#section-5.1
+        let mut link_header = match link_split.next() {
+            None => return Err(::Error::Header),
+            Some(link_value) => match verify_and_trim(link_value, '<', '>') {
+                Err(_) => return Err(::Error::Header),
+                Ok(link_url) => match Uri::new(link_url) {
+                    Err(_) => return Err(::Error::Header),
+                    Ok(uri) => Link {
+                        link: uri,
+                        rel: None,
+                        anchor: None,
+                        rev: None,
+                        href_lang: Vec::new(),
+                        media_desc: None,
+                        title: None,
+                        title_star: None,
+                        media_type: None,
+                        link_extension: None,
+                    }
+                },
+            },
+        };
+
+        // Parse the Link header's parameters
+        for link_param in link_split {
+            let mut link_param_split = link_param.split('=');
+
+            let link_param_name = match link_param_split.next() {
+                None => return Err(::Error::Header),
+                Some(p) => p.trim(),
+            };
+
+            match link_param_name.to_lowercase().as_ref() {
```

Comment:
I have tried `UniCase` before doing the above. Maybe there's a better way I will check.